tsp-output/@azure-tools/typespec-csharp/azoai/AzureOpenAI/AzureChatClient.cs

```diff
@@ -9,29 +10,41 @@ internal class AzureChatClient : Chat
 {
     private readonly string _apiVersion;
 
-    // TODO/Note: doesn't need to be virtual -- derived client could also set it in the 
-    // constructor.  Is one approach better?  I guess setting it in the constructor
-    // requires making it settable which maybe we don't want.
-    protected override ModelReaderWriterOptions ModelReaderWriterOptions => new("AzureWire");
-
     internal AzureChatClient(ClientPipeline pipeline, ApiKeyCredential credential, Uri endpoint, string apiVersion)
         : base(pipeline, credential, endpoint)
     {
         _apiVersion = apiVersion;
     }
 
-    public override Task<ClientResult<CreateChatCompletionResponse>> CreateChatCompletionAsync(CreateChatCompletionRequest createChatCompletionRequest, CancellationToken cancellationToken = default)
+    public override async Task<ClientResult<CreateChatCompletionResponse>> CreateChatCompletionAsync(CreateChatCompletionRequest createChatCompletionRequest, CancellationToken cancellationToken = default)
     {
-        return base.CreateChatCompletionAsync(createChatCompletionRequest, cancellationToken);
+        Argument.AssertNotNull(createChatCompletionRequest, nameof(createChatCompletionRequest));
+
+        // Initialize Azure input model
+        AzureCreateChatCompletionRequest azureChatCompletionOptions = new(createChatCompletionRequest);
```

Comment:
Could you just wrap and then call the base implementation? It would be good to not have to reimplement the convenience methods totally from scratch.

Reply:
I think when the protocol method signature is the same that will work.  But since here we needed to add an overload for the protocol method because an input value has moved from the request body into a path parameter, we need to reimplement it to call the new protocol method overload.

Reply:
Ah, makes sense. But, where is the new protocol method? I have a hard time finding it. I was interested in seeing how you think we will make the new protocol method available to the callers.

Reply:
OK, I found the protocol method. But, the method needs to be an extension method, not just a public method on the internal type. People won't be able to call it, if it's an instance method

Reply:
Good point!  That is my mistake.